### PR TITLE
Refactor Apparmor Profile generation

### PR DESCRIPTION
**Pull request by daemon1024**
We now have a way to resolve conflicts between local and global profile allowing us to set a priority order in the code for what to expect when conflicting rules exist between a global policy and fromSource.

This conflict deemed a complex resolution logic so I ended up refactoring the profile generation logic to more freely analyse and manipulate the profiles.

Will handle in BPF LSM in separate PR

Ref #787

### CODE REVIEW

File: KubeArmor/enforcer/appArmorTemplate.go [hunk 1]
```diff
@@ -0,0 +1,202 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2021 Authors of KubeArmor
```
Comment (nyrahul) [inline KubeArmor/enforcer/appArmorTemplate.go]:
```suggestion
// Copyright 2022 Authors of KubeArmor
```

**Review by nyrahul — COMMENTED** [review]
This is a Photogenic PR! LGTM

**Comment by achrefbensaad** [comment]
LGTM!

**Review by achrefbensaad — APPROVED** [review]

**Comment by achrefbensaad** [comment]
@daemon1024 Can you please squash your commits.

**Comment by daemon1024** [comment]
> @daemon1024 Can you please squash your commits.

@achrefbensaad The commits address different problems individually, would prefer to keep it this way!

**Review by nyrahul — APPROVED** [review]
LGTM

**Pull request merged**